tests/components/backup/test_manager.py

```diff
@@ -534,21 +536,75 @@ async def test_loading_platforms(
 
     assert not manager.platforms
 
+    get_agents_mock = AsyncMock(return_value=[])
+
     await _setup_backup_platform(
         hass,
         platform=Mock(
             async_pre_backup=AsyncMock(),
             async_post_backup=AsyncMock(),
-            async_get_backup_agents=AsyncMock(),
+            async_get_backup_agents=get_agents_mock,
         ),
     )
     await manager.load_platforms()
     await hass.async_block_till_done()
 
     assert len(manager.platforms) == 1
-
     assert "Loaded 1 platforms" in caplog.text
 
+    get_agents_mock.assert_called_once_with(hass)
+
+
+class LocalBackupAgentTest(BackupAgentTest, LocalBackupAgent):
+    """Local backup agent."""
+
+    def get_backup_path(self, backup_id: str) -> Path:
+        """Return the local path to a backup."""
+        return "test.tar"
+
+
+@pytest.mark.parametrize(
+    ("agent_class", "num_local_agents"),
+    [(LocalBackupAgentTest, 1), (BackupAgentTest, 0)],
+)
+async def test_loading_platform_with_listener(
+    hass: HomeAssistant,
+    agent_class: type[BackupAgentTest],
+    num_local_agents: int,
+) -> None:
+    """Test loading a backup agent platform which can be listened to."""
+    manager = BackupManager(hass, CoreBackupReaderWriter(hass))
+
+    assert not manager.platforms
+
+    get_agents_mock = AsyncMock(return_value=[agent_class("remote1", backups=[])])
+    register_listener_mock = Mock()
+
+    await _setup_backup_platform(
+        hass,
+        domain="test",
+        platform=Mock(
+            async_get_backup_agents=get_agents_mock,
+            async_register_backup_agents_listener=register_listener_mock,
+        ),
+    )
+    await manager.load_platforms()
+    await hass.async_block_till_done()
+    assert list(manager.backup_agents) == ["test.remote1"]
```

Comment:
Can we do the agent assertion on the response from the websocket command `"backup/agents/info"` instead? Then I think we can rewrite the test to not use the manager directly.